src/plugins/intel_cpu/src/emitters/x64/jit_load_store_emitters.cpp

```diff
@@ -97,6 +97,24 @@ size_t jit_load_emitter::aux_gprs_count() const {
 }
 
 void jit_load_emitter::emit_impl(const std::vector<size_t> &in_idxs, const std::vector<size_t> &out_idxs) const {
+    // save runtime debug info
+    h->push(h->r15);
+    Xbyak::Label label_set_current;
+    h->mov(h->r15, reinterpret_cast<size_t>(&start_address));
+    h->cmp(h->qword[h->r15], 0);
+    h->jne(label_set_current);
+    h->mov(h->qword[h->r15], Xbyak::Reg64(in_idxs[0]));
+    h->L(label_set_current);
+    {
+        h->mov(h->r15, reinterpret_cast<size_t>(&current_address));
+        h->mov(h->qword[h->r15], Xbyak::Reg64(in_idxs[0]));
+
+        // iteration++
+        h->mov(h->r15, reinterpret_cast<size_t>(&iteration));
+        h->add(h->qword[h->r15], 0x01);
+    }
+    h->pop(h->r15);
```

Comment:
Do I understand correctly that we need this code only for debug info printing? If yes, I don't think that we always need to perform it. We probably need a mechanism which allows us to switch on debug mode in emitters. Maybe we can introduce some `enum`/`flag` in `jit_emitters` class?
@IvanNovoselov @chenhu-wang what do you think?

Reply:
@v-Golubev , @IvanNovoselov Regarding passing enable_snippets_err_detector flag to emitters. I see 2 ways:
1. update each emitter constructor to set the flag as a member. will update each emitters constuctor such as [add_emitter](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/x64/jit_eltwise_emitters.cpp)
2. add flag as a parameter to emit_code() directly. pass from [entrance ](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/generator.cpp#L39)to each emitters.
Both need update lot of interfaces. As this is not a functional feature, to keep API clean, I use a global flag similar like error_handler to indicate enable or disable. Welcome comment if other suggestions. Thanks!
global flag:
https://github.com/openvinotoolkit/openvino/pull/19935/files#diff-3c0cab6cbe926348e49ee79a66ea4c04056b8c2aff2d45b056814ae902176fbeR15
https://github.com/openvinotoolkit/openvino/pull/19935/files#diff-57def2b783e8aa688a1e057c0f2d8b65f358e820a42fa2f61439d6b6201683c4R23

Reply:
I see 3rd option here: we can have base class member which will be `false` by default, but this can be changed in constructor if the corresponding environment variable is set. We already have some [CPU specific environment variables](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/docs/debug_capabilities/README.md) which regulate debug settings. We can add a new one there.

Additional notice: this env variable must only be taken into account when `CPU_DEBUG_CAPS` are enabled.

Reply:
ok, this is more align with other debug capability, done, thanks! [71c3b32](https://github.com/openvinotoolkit/openvino/pull/19935/commits/71c3b32e671920e548fecc71cf3705ca80ed7ecf)